temporal-sdk/src/main/java/io/temporal/client/UpdateWithStartWorkflowOperation.java

```diff
@@ -0,0 +1,413 @@
+   *
+   * @param request method reference to a proxy created through {@link
+   *     WorkflowClient#newWorkflowStub(Class, WorkflowOptions)}.
+   * @param arg1 first request function parameter
+   * @param arg2 second request function parameter
+   * @param arg3 third request function parameter
+   * @param arg4 fourth request function parameter
+   */
+  public static <A1, A2, A3, A4> Builder<Void> newBuilder(
+      Functions.Proc4<A1, A2, A3, A4> request, A1 arg1, A2 arg2, A3 arg3, A4 arg4) {
+    return new Builder<>(() -> request.apply(arg1, arg2, arg3, arg4));
+  }
+
+  /**
+   * Returns a new builder for an {@link UpdateWithStartWorkflowOperation} for a five argument
+   * request.
+   *
+   * @param request method reference to a proxy created through {@link
+   *     WorkflowClient#newWorkflowStub(Class, WorkflowOptions)}.
+   * @param arg1 first request function parameter
+   * @param arg2 second request function parameter
+   * @param arg3 third request function parameter
+   * @param arg4 fourth request function parameter
+   * @param arg5 fifth request function parameter
+   */
+  public static <A1, A2, A3, A4, A5> Builder<Void> newBuilder(
+      Functions.Proc5<A1, A2, A3, A4, A5> request, A1 arg1, A2 arg2, A3 arg3, A4 arg4, A5 arg5) {
+    return new Builder<>(() -> request.apply(arg1, arg2, arg3, arg4, arg5));
+  }
+
+  /**
+   * Returns a new builder for an {@link UpdateWithStartWorkflowOperation} for a six argument
+   * request.
+   *
+   * @param request method reference to a proxy created through {@link
+   *     WorkflowClient#newWorkflowStub(Class, WorkflowOptions)}.
+   * @param arg1 first request function parameter
+   * @param arg2 second request function parameter
+   * @param arg3 third request function parameter
+   * @param arg4 fourth request function parameter
+   * @param arg5 fifth request function parameter
+   * @param arg6 sixth request function parameter
+   */
+  public static <A1, A2, A3, A4, A5, A6> Builder<Void> newBuilder(
+      Functions.Proc6<A1, A2, A3, A4, A5, A6> request,
+      A1 arg1,
+      A2 arg2,
+      A3 arg3,
+      A4 arg4,
+      A5 arg5,
+      A6 arg6) {
+    return new Builder<>(() -> request.apply(arg1, arg2, arg3, arg4, arg5, arg6));
+  }
+
+  /**
+   * Returns a new builder for an {@link UpdateWithStartWorkflowOperation}.
+   *
+   * @param updateName name of the update handler
+   * @param resultClass class of the update return value
+   * @param args update request arguments
+   */
+  public static <R> Builder<R> newBuilder(String updateName, Class<R> resultClass, Object[] args) {
+    return new Builder<>(
+        UpdateOptions.<R>newBuilder().setResultClass(resultClass).setUpdateName(updateName), args);
+  }
+
+  private WorkflowStub stub;
+
+  private UpdateOptions<R> options;
+
+  private Object[] updateArgs;
+
+  private final CompletableFuture<WorkflowUpdateHandle<R>> handle;
+
+  private final Functions.Proc request;
+
+  public UpdateWithStartWorkflowOperation(
```

Comment:
Why does this need to be public? Generally if a class has a builder the constructor is private so users have to use the builder.

Reply:
👍 good catch; made it private now